mitodl/ocw-hugo-themes
# Initial commit

#### What's this PR do?
Currently, the OCW website is built out of 2 main components.  `ocw-www` is a Hugo site with its own webpack build that contains content that builds the home page, search page, testimonials, promos, etc.  `ocw-course-hugo-theme` is a Hugo theme used to build the course sites in the website.  The problem with this is that these sites share a lot of the same styles and some templates, so changes to similar components have to be done twice in both repos.

This PR is the initial commit to a new repo combining the responsibilities of `ocw-www` and `ocw-course-hugo-theme` into a single module based Hugo theme that can be inherited by any Hugo site.

#### Where should the reviewer start?
Read the updated [`README.md`](https://github.com/mitodl/ocw-hugo-theme/blob/1cd4142fc0fca34b756ed3538a7851608c494a73/README.md) file to familiarize yourself with how this works

#### How should this be manually tested?
- Clone [`ocw-www` ](https://github.com/mitodl/ocw-www) and check out the `cg/content-only` branch
- Follow the updated readme and:
  - Install dependencies and configure the site based on the instructions
  - When you create your `.env` file, set `EXTERNAL_SITE_PATH` to the path to `ocw-www` that you cloned earlier
  - Edit the `go.mod` in your `ocw-www` folder to override it to point at your local copy of `ocw-hugo-theme` by adding the following lines, replacing the paths with yours:
```
replace github.com/mitodl/ocw-hugo-theme/base-theme => /path/to/ocw-hugo-theme/base-theme

replace github.com/mitodl/ocw-hugo-theme/www => /path/to/Code/ocw-hugo-theme/www

replace github.com/mitodl/ocw-hugo-theme/course=> /path/to/Code/ocw-hugo-theme/course
```
  - Spin up the OCW home page locally using `npm start` and verify that everything looks alright when compared to https://ocwnext.odl.mit.edu/
  - Visit search and verify that everything looks like the prod site
  - Spin up a course site using `npm run start:course` and verify that the course site looks as it does on https://ocwnext.odl.mit.edu/
